### DISCUSSION TITLE
add_html_dependency not in head for filters

### DISCUSSION THREAD

**Discussion by mloubout** [quarto-dev/quarto-cli#5433] [Feature Requests]
Hi

I am working on a lua filter and one piece I would need is to be able to do `add_html_dependency` for a javascript but to be added in the body, not the head. Currently I am bypassing it by plainly adding the content. of the script as text after the body but It would be nice to be able to just include it.

**Comment by cderv** [comment]
From Lua you have also the include API to add some content in different place of the template. `before-body` or `after-body` is an option. 
You'll need to do some processing that add_html_dependency is doing for you to create the right html code to insert, but it should help though. 

See https://quarto.org/docs/extensions/lua-api.html#includes

**Comment by mloubout** [comment]
Yes that's what I'm currently using by doing using `quarto.doc.include_text` and I have a plain raw HTML added as text after the body.  It's a small script so it's fine just adding it plain but might be good to think of nice ways to do it for fancy ones.

Another "bug" related to the same API `add_html_dependency` doesn't seem to accept URLs as inputs for CSS/scripts, i.e something like

```
  quarto.doc.add_html_dependency({
    scripts = {'some-js-url'},
    stylesheets = {'some-css-url'}
  })
```

unless I missed part of the documentation

**Reply by cderv** [reply]
> It's a small script so it's fine just adding it plain but might be good to think of nice ways to do it for fancy ones.

the `include_file()` would help with fancy one, wouldn't it ? 

Can you give an example of a html dependency requiring adding also some content in body ? 

> Another "bug" related to the same API add_html_dependency doesn't seem to accept URLs as inputs for CSS/scripts,

You did not miss something. I believe it is not a feature we wanted to make easy to do because of potential vulnerability: See https://cwe.mitre.org/data/definitions/830.html
